### PR TITLE
Stack Overflow Error Search Auto Tool

### LINKED ISSUE #1063: Script for Auto Search Tool for Stack Overflow
# Aim

- Automating the process of searching errors on stackoverflow.

# Details

- Script will track errors and use the stackexchange api for getting a json response about the questions related on the Stack Overflow website.
- Parse it and open some links of stackoverflow threads on the web browser.

### PR DESCRIPTION
**Pull request by Muskan0**
# Description

I added the Stack Overflow Tool. It will execute a python script, if there will be any errors, some of the related Stack Overflow questions will be opened in the browser.

## Type of change

_Please delete options that are not relevant._

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Update

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] My documentation follows [`Template for README.md`](https://github.com/avinashkranjan/Amazing-Python-Scripts/blob/master/Template%20for%20README.md)
- [x] I have added the project meta data in the PR template.
- [x] I have created the ``requirements.txt`` file if needed.

## Project Metadata

`` If there is no-file/nothing to fill the below fields with, then type: none ``

`` Example: `` If no requirements.txt needed/present, then type none in ``Requirments``.

Category:
- [ ] Calculators
- [ ] AI/ML
- [ ] Scrappers
- [ ] Social_Media
- [ ] PDF
- [ ] Image_Processing
- [ ] Video_Processing
- [ ] Games
- [ ] Networking
- [ ] OS_Utilities
- [x] Automation
- [ ] Cryptography
- [ ] Computer_Vision
- [ ] Fun
- [ ] Others

Title: Stack Overflow Error Search Auto Tool

Folder: Stackoverflow-Tool

Requirements: requirements.txt

Script: script.py

Arguments: None

Contributor: Muskan0

Description: This script executes your python file and it opens up the related Stack Overflow threads in the browser if there's any error in the script.

### CODE REVIEW

File: Stackoverflow-Tool/filename.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# this is the python script taken as an example
```
Comment (kaustubhgupta) [inline Stackoverflow-Tool/filename.py:1]: you can remove this file

File: Stackoverflow-Tool/script.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import subprocess
+import requests
+import webbrowser
+
+# executing a script and extracting the errors
+p = subprocess.run(['python', 'filename.py'], capture_output=True, text=True)
```
Comment (kaustubhgupta) [inline Stackoverflow-Tool/script.py:6]: get the file name or exact path from the user

File: Stackoverflow-Tool/README.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+Stack Overflow Error Search Auto Tool
+
+- This script executes your python file.
+- And it opens up the related Stack Overflow threads in the browser if there's any error in the script.
+
+## Setup instructions
+
+- In the script add the required python file name which you want to be executed and checked for errors.
+- Add the name of the required python file in place of filename.py as given in the script.
+- Install these python libraries: subprocess, requests and webbrowser.
```
Comment (kaustubhgupta) [inline Stackoverflow-Tool/README.md:12]: you can remove this section

**Review by kaustubhgupta — CHANGES_REQUESTED** [review]

**Comment by Muskan0** [comment]
@kaustubhgupta I did the changes, please review it.

**Review by kaustubhgupta — APPROVED** [review]
good

**Review by santushtisharma10 — APPROVED** [review]

**Review by avinashkranjan — APPROVED** [review]
LGTM..💯

**Pull request merged**